File: programmers/Level 1/incomplete_player.cpp
/*
programmers Level 1: 완주하지 못한 선수
2022-03-23
*/
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

string solution(vector<string> participant, vector<string> completion) {
    string answer = "";
    
    sort(participant.begin(), participant.end());
    sort(completion.begin(), completion.end());
    
    for(int i = 0 ; i < participant.size() ; i++){
        if(completion[i] != participant[i] || i == participant.size() - 1){
            answer = participant[i];
            break;
        }
    }
    
    return answer;
}

File: programmers/Level 1/incomplete_player.py
# programmers Level 1: 완주하지 못한 선수
# date: 2022-04-15
def solution(participant, completion):
    answer = ''
    dic = {}
    
    for x in participant:
        dic[x] = dic.get(x, 0) + 1
    
    for x in completion:
        dic[x] -= 1
    
    for k, v in dic.items():
        if v == 1:
            answer = k
            break
            
    return answer